Signal/src/util/OWSScreenLock.swift

```diff
@@ -0,0 +1,334 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import LocalAuthentication
+
+@objc public class OWSScreenLock: NSObject {
+
+    let TAG = "[OWSScreenLock]"
+
+    public enum OWSScreenLockOutcome {
+        case success
+        case cancel
+//        case userCancel
+//        case otherCancel
```

Comment:
I'm going to come back to this in the next PR.  I want to make sure we're handling all edge cases properly.